### PR TITLE
feat(cr, wk): make clicks, input and evaluate await scheduled navigations

### PR DESCRIPTION
**Pull request by pavelfeldman**

### CODE REVIEW

File: test/navigation.spec.js [hunk 1]
```diff
@@ -800,6 +800,93 @@ module.exports.describe = function({testRunner, expect, playwright, MAC, WIN, FF
...
+    it('should not throw when clicking on links which do not commit navigation', async({page, server, httpsServer}) => {
+      await page.goto(server.EMPTY_PAGE);
+      await page.setContent(`<a href='${httpsServer.EMPTY_PAGE}'>foobar</a>`);
+      await page.click('a');
+    });
+    it('should not throw when clicking on download link', async({page, server, httpsServer}) => {
+      await page.setContent(`<a href="${server.PREFIX}/wasm/table2.wasm" download=true>table2.wasm</a>`);
+      await page.click('a');
+    });
+    it.fail(CHROMIUM || FFOX)('should not hang on window.stop', async({page, server, httpsServer}) => {
```
Comment (JoelEinbinder) [inline test/navigation.spec.js]: I don't feel comfortable landing this patch until this test passes.

Comment (pavelfeldman) [reply test/navigation.spec.js]: That's too bad, because it does not pass without my patch either.

Comment (pavelfeldman) [reply test/navigation.spec.js]: Figured it out!

Comment (JoelEinbinder) [reply test/navigation.spec.js]: Interesting! Something stalled the evaluation?

**Review by dgozman — APPROVED** [review]

**Pull request merged**